interfaces/apparmor/spec.go

```diff
@@ -48,6 +49,46 @@ func (spec *Specification) AddSnippet(snippet string) {
 	}
 }
 
+func snippetFromLayout(layout *snap.Layout) string {
+	mountPoint := layout.Snap.ExpandSnapVariables(layout.Path)
+	return fmt.Sprintf("# Layout path: %[1]s\n%[1]s{,/**} mrwklix,", mountPoint)
+}
+
+// AddSnapLayout adds apparmor snippets based on the layout of the snap.
+func (spec *Specification) AddSnapLayout(si *snap.Info) {
+	if len(si.Layout) == 0 {
+		return
+	}
+
+	// walk the layout elements in deterministic order, by mount point name
+	paths := make([]string, 0, len(si.Layout))
+	for path := range si.Layout {
+		paths = append(paths, path)
+	}
+	sort.Strings(paths)
+
+	// get tags describing all apps and hooks
+	tags := make([]string, 0, len(si.Apps)+len(si.Hooks))
+	for _, app := range si.Apps {
+		tags = append(tags, app.SecurityTag())
+	}
+	for _, hook := range si.Hooks {
+		tags = append(tags, hook.SecurityTag())
+	}
+
+	// append layout snippets to all tags
```

Comment:
An expanded comment here would be nice. Ie, mentioning that layout are intended to be snap wide...

Reply:
Ack, will do

Reply:
Blocker: please add add this comment.